### DISCUSSION TITLE
⚡ Gutenberg is a very poorly designed editor, here’s why.

### DISCUSSION THREAD

**Discussion by vovkasolovev** [WordPress/gutenberg#41830] [Developer Experience]
1. The worst part: **Gutenberg is not a WYSIWYG editor**. 
It has its own styles that have nothing to do with the appearance of the site.  Gutenberg has its own column width, indentation, font and color settings. Gutenberg has its own HTML structure. The site’s theme should have separately specified styles for the Gutenberg editor that would match the site’s style exactly.
This makes it very difficult to make the styles look the same. With the Classic Editor you can have the same CSS styles for the Website and for the Classic Editor. But for the Gutenberg editor you have to create styles for a different structure that look exactly like the Site styles. It is impossible for the theme developer to maintain this exactly. 

2. **A lot of unclear states.**
Absent or weak indication of the selected object. Due to the fact that in the left column grouped Adding new blocks and List of blocks, and in the right column grouped Post and Block settings is not always clear what we’re working with – with a block, the structure or post settings.
It would be better to never hide the Block List, and in another column group Add New or Selected Block Settings. And remove Post settings from the tabs with the Block settings - these are settings from another level, not alternative.

3. **The interface blinks and changes location.**
For example, after selecting a block, a Settings container appears above it that overlaps the block above. Another example, when you select a block, it may enlarge, showing some hidden options that for some reason aren’t placed in the right column of the block’s settings. And a very unpleasant example: constantly closing the List of used blocks on the left, causing the field with the main content to expand, changing the view and location.

4. **Uncompact interface.**
Only a few elements fit into the height of the screen. Very large labels above the fields, large indents between the fields, large fonts. The settings are grouped in Accordion of large heights. Often it is not possible to see all settings at once. As a result it requires a lot of scrolling.

5. **Excessive number of icons.**
Very many actions are encoded in pictograms. It is not at all clear what all those different kinds of arrows and sticks are responsible for.

6. **Too many blocks.**
There are different blocks with little variation in features. Widget blocks and structure blocks are in the same pile. Lots of rarely used blocks. Blocks can be disabled – to do this, you have to click on each one in a separate menu.

7. **Lack of common and basic style settings for blocks.**
Example: indents, colors, background, sizes. Since there are many blocks, not all are designed to display well in any theme. This not infrequently forces you to customize the appearance of the block through additional CSS.

8. **Not designed for plugins.**
Editor plugins often add a large number of customizations. In the Classic Editor, these settings were placed at the bottom in the wide part of the editor. In the Gutenberg, some plug-in settings can only be placed as one of the Accordion elements in the narrow column on the right. As a result, the narrow column is overloaded and long.
The user needs to be able to organize the interface added by plugins into their own columns, tabs or groups. Now you can only rearrange the settings in up-down order.

**Comment by annezazu** [comment]
@vovkasolovev You're touching on a ton here in this discussion thread! I appreciate you taking the time to summarize your high level feedback. If you're game, I'd highly recommend splitting what you can into bug reports (ex: "The interface blinks and changes location" section sounds like bug reports), general issues that you'd like to see improved (ex: "Not designed for plugins." what would you like to see here?), and then joining current discussions (ex: "Lack of common and basic style settings for blocks" aka issues like https://github.com/WordPress/gutenberg/issues/28356 and https://github.com/WordPress/gutenberg/issues/41113). It would be really helpful to have specifics in the right places from the above either way. For a touch more info: 

>Very many actions are encoded in pictograms. It is not at all clear what all those different kinds of arrows and sticks are responsible for.

You might enjoy the Text Only Icons by the way: https://make.wordpress.org/core/2020/12/21/core-editor-improvement-text-only-icons-for-the-toolbar/

>This makes it very difficult to make the styles look the same. With the Classic Editor you can have the same CSS styles for the Website and for the Classic Editor. But for the Gutenberg editor you have to create styles for a different structure that look exactly like the Site styles. It is impossible for the theme developer to maintain this exactly.

You might like digging into this post/effort address CSS rendered by the block editor: https://make.wordpress.org/core/2022/06/24/block-editor-styles-initiatives-and-goals/

**Reply by vovkasolovev** [reply]
You noted in my words only what can be solved by breaking into small edits, breaking into separate bug reports and fixing by adding features. That's a good way to solve problems when you're on the right track. But first of all, that's a small part of the problems I wrote about. And secondly, these problems wouldn't exist if there wasn't a larger problem.

The problem is extensive and conceptual - the amount of features inherent in the Gutenberg interface ended up at the level of a block text editor, when it's important to have a small number of features and no WYSIWYG. With the addition of features, Gutenberg has turned into a site editor with more features, and there are plans to cram even more in, but in the original wrapper such features no longer fit and such an editor should already be WYSIWYG and should be careful to focus the user's attention. 

Can you please advise me where to discuss the visual layout of the Gutenberg interface, discuss the architecture of code wrap and draw a redesign?

**Reply by annezazu** [reply]
>But first of all, that's a small part of the problems I wrote about. And secondly, these problems wouldn't exist if there wasn't a larger problem.

Totally hear you there! I would take great inspiration from efforts like this post: https://mrwweb.com/standardized-design-tokens-css-wordpress-future/ which led to this overall issue https://github.com/WordPress/gutenberg/issues/38998 and various sub-issues, including one that's in the process of being addressed/explored in depth: https://github.com/WordPress/gutenberg/issues/39371#issuecomment-1153457475

I'd recommend getting specific about the bugs and reporting those where you can since those can ideally be resolved in the short term. From there, for the larger convo, I'd recommend an overarching proposal for what you'd like to discuss shared in the open, whether on your blog or perhaps as a follow up here. In terms of the editor being WYSIWYG, that's very much the intention and built into the entire premise of the Site Editor so I think that's why I responded with the route of filing bugs :)  You might enjoy these overview posts to get a deeper sense of things: https://developer.wordpress.org/block-editor/getting-started/outreach/#overviews-of-gutenberg Here are a few more recent items not mentioned there:  [an issue on unifying the interface](https://github.com/WordPress/gutenberg/issues/41717) and a post on thinking [through the future of the WordPress admin experience.](https://make.wordpress.org/design/2022/06/13/thinking-through-the-wordpress-admin-experience/)

**Reply by astralliquid** [reply]
I don't the the OP even need to report to any relevant team. Any Gutenberg developer with some common sense would have seen the problem or wouldn't have coded it as such. The OP is just highlighting all the problems that are solvable with common sense. There is no point reporting to everywhere trying to tell them there is a bigger problem. Nobody can understand how to use all those icons and empty spaces and lack of intuition. If WP need users to report such trivial things, you can see how technical and inconsiderate the head behind the project. Steve Job or Apple would have fire the entire UX team.
.
